src/extensions/manifest.ts

```diff
@@ -56,6 +57,44 @@ export async function writeToManifest(
 
   writeExtensionsToFirebaseJson(specs, config);
   await writeEnvFiles(specs, config, options.force);
+  await writeLocalSecrets(specs, config, options.force);
+}
+
+// TODO(lihes): Add unit tests.
+async function writeLocalSecrets(
+  specs: ManifestInstanceSpec[],
+  config: Config,
+  force?: boolean
+): Promise<void> {
+  for (const spec of specs) {
+    if (!spec.paramSpecs) {
+      continue;
+    }
+
+    let writeBuffer: Record<string, string> = {};
+    for (const paramSpec of spec.paramSpecs.filter((p) => p.type === ParamType.SECRET)) {
```

Comment:
Nit: I think this code ends up cleaner if we also filter for only params that have a local override- ie:
```
const locallyOverridenSecretParams = spec.paramsSpecs.filter((p) => p.type === ParamType.Secret && spec.params[p.param].local)
```
```

Reply:
good suggestion, done